### ISSUE TITLE
Softmax Splatting for Video Frame Interpolation

### ISSUE THREAD

**Issue by reyllama** [reyllama/paper-reviews#94]
*CVPR 2020*

## Motivation

- Backward warping based method has been widely studied, but has the shortcoming of having to approximate reverse optical flow. 
- Overcome the innate ambiguity of forward-warping based image sampling through softmax splatting.

## Key Idea

- Forward warping incurs ambiguity of more than one pixel being mapped to a single discrete coordinate.
- Softmax splatting is translation invariant w.r.t. depth (or importance metric) and thus resolves the ambiguity.

## Previous Approach

- Backward warping has been an active topic of research
- Kernel based, Flow based, Phase based

### Splatting (Forward) vs Sampling (Backward)

![image](https://user-images.githubusercontent.com/46308407/128817964-bc9166a7-b839-478a-a4dd-33626a57ffdb.png)

- Backward warping is free from innate ambiguity, while has the trouble of approximating reverse optical flow. (Inhibits the use of *off-the-shelf* optical flow models)
- Forward warping is straight-forward, but multiple pixels can be mapped to the same discrete coordinate, which causes ambiguity.

### Different Splatting Operators
![image](https://user-images.githubusercontent.com/46308407/128819259-25bcbd7c-eb39-49a8-b980-3f34dee2b29f.png)
- Summation : brightness inconsistency as pixel activations are added
- Average : brightness is consistent but overlapping regions are simply averaged (ghosting)
- Linear : Not invariant to translation (ex: depth (1/1, 1/10) vs (1/101, 1/110) - latter is nearly averaged)
- Softmax : Invariant to translation and affected by scaling
![image](https://user-images.githubusercontent.com/46308407/128818485-5edb233a-e608-4118-99f7-5bfed945e2fb.png)

### Importance Metric
- Computing depth for each pixel is computationally prohibiting, hence propose an alternative importance metric for softmax splatting.
![image](https://user-images.githubusercontent.com/46308407/128818558-74a8a6f1-e4bb-4d1c-8619-7a97e3239399.png)
- Can be further fine-tuned using a small neural network v.
![image](https://user-images.githubusercontent.com/46308407/128818564-bd19e020-98c3-4eeb-a8e2-61a9f5b05baa.png)

### Feature Pyramid
![image](https://user-images.githubusercontent.com/46308407/128818830-0ebe6ddb-575d-4ebe-b82e-7fe89f1cc99d.png)
![image](https://user-images.githubusercontent.com/46308407/128818842-eaa2fd91-5562-4f6e-bd96-e0e96abfa4fc.png)
- As softmax splatting is differentiable, the whole model can be trained end-to-end.
- Extract multi-resolution feature maps from a convolutional feature extractor.
![image](https://user-images.githubusercontent.com/46308407/128818797-b86d34be-26ca-4a95-882c-a906e3ea008e.png)
- Image synthesis network is U-Net like GridNet.
- Importance metric is refined using a 3-level U-Net.

### Dataset
- Vimeo-90K
- Middlebury
- UCF101
- Xiph

### Baseline Comparison
![image](https://user-images.githubusercontent.com/46308407/128819285-aa769aa4-cc14-4fa5-b608-cabe103fc735.png)
![image](https://user-images.githubusercontent.com/46308407/128819050-cbfffe6a-220a-4752-a11d-5c0c6e454df8.png)
![image](https://user-images.githubusercontent.com/46308407/128819077-2a75acd4-4f23-4dc7-acd5-baf52c5e5f86.png)

### Ablation
![image](https://user-images.githubusercontent.com/46308407/128819112-6233c0de-a360-47b8-94c3-f49c0f26b9fc.png)

## Comments
- Flow-based frame interpolation.
- Forward warping and backward warping. Notes taken.
- Not much experiments. Not even visual outputs much provided.
